libraries/rush-lib/src/logic/pnpm/SubspacePnpmfileConfiguration.ts

```diff
@@ -86,4 +94,92 @@ export class SubspacePnpmfileConfiguration {
+    const subspaceProjectsMap: Map<string, RushConfigurationProject> = new Map();
+    for (const project of rushConfiguration.projects) {
+      if (subspace.contains(project)) {
+        subspaceProjectsMap.set(project.packageName, project);
+      } else {
+        workspaceProjectsMap.set(project.packageName, project);
+      }
+
+      projectNameToInjectedDependenciesMap.set(project.packageName, new Set());
+    }
+
+    const processTransitiveInjectedInstallQueue: Array<RushConfigurationProject> = [];
+
+    for (const subspaceProject of subspaceProjectsMap.values()) {
+      const dependenciesMeta: IDependenciesMetaTable | undefined =
+        subspaceProject.packageJson.dependenciesMeta;
+      for (const dependencyName in dependenciesMeta) {
+        if (dependenciesMeta[dependencyName]?.injected) {
+          projectNameToInjectedDependenciesMap.get(subspaceProject.packageName)?.add(dependencyName);
+
+          //if this dependency is in the same subspace, leave as it is, PNPM will handle it
+          //if this dependency is in another subspace, then it is transitive injected installation
+          //so, we need to let all the workspace dependencies along the dependency chain to use injected installation
+          if (!subspaceProjectsMap.has(dependencyName)) {
+            processTransitiveInjectedInstallQueue.push(workspaceProjectsMap.get(dependencyName)!);
+          }
+        }
+      }
+    }
+
+    // rewrite all workspace dependencies to injected install all for transitive injected installation case
+    while (processTransitiveInjectedInstallQueue.length > 0) {
+      const currentProject: RushConfigurationProject | undefined =
+        processTransitiveInjectedInstallQueue.shift();
+      const dependencies: Record<string, string> | undefined = currentProject?.packageJson?.dependencies;
+      const devDependencies: Record<string, string> | undefined =
+        currentProject?.packageJson?.devDependencies;
+
+      if (currentProject && dependencies) {
+        SubspacePnpmfileConfiguration._processDependenciesForTransitiveInjectedInstall(
+          projectNameToInjectedDependenciesMap,
+          processTransitiveInjectedInstallQueue,
+          dependencies,
+          currentProject,
+          rushConfiguration
+        );
+      }
+
+      if (currentProject && devDependencies) {
+        SubspacePnpmfileConfiguration._processDependenciesForTransitiveInjectedInstall(
+          projectNameToInjectedDependenciesMap,
+          processTransitiveInjectedInstallQueue,
+          devDependencies,
+          currentProject,
+          rushConfiguration
+        );
+      }
+    }
+
+    return projectNameToInjectedDependenciesMap;
+  }
+
+  private static _processDependenciesForTransitiveInjectedInstall(
+    projectNameToInjectedDependencies: Map<string, Set<string>>,
+    processTransitiveInjectedInstallQueue: Array<RushConfigurationProject> = [],
```

Comment:
When wouldn't this be provided?

Reply:
Sorry, I did not get the point here. What is not provided here?

Reply:
I think Ian's point is that this parameter doesn't need to be optional.  Also, optional parameters generally should be avoided except in public APIs with heavy usage.  For example:

```ts
function doOtherThing(url: string, workingFolder?: string): void {
}

function doThing(url: string, workingFolder?: string): void {
  doOtherThing(url); // <- mistake
}
```

This mistake won't happen with `string | undefined`.